security/pkg/server/ca/authenticate/kube_jwt.go

```diff
@@ -31,6 +34,9 @@ const (
 	KubeJWTAuthenticatorType = "KubeJWTAuthenticator"
 )
 
+// TODO: move this under pkg/k8s - it depends on k8s client library.
+// The cert and OIDC are independent of k8s.
```

Comment:
Not sure how this is independent of k8s - there is a differnt OIDC authenticator, isn't this one explicitly for k8s?

Reply:
I mean: this go package has k8s, oidc and certs. The last 2 have no dep on K8S client libraries ( which are large ). 
I think it would be better to have the k8s-related plugins in one of the 2 main k8s packages - probably in the 'registry' one since it's multi-cluster aware. Will save a lot of init and duplication. 

Istiod can also run without k8s dependency - using MCP or other stores - in which case we wouldn't link k8s libs.